invoices/invoiceregistry.go

```diff
@@ -1636,13 +1593,15 @@ func (i *InvoiceRegistry) SubscribeSingleInvoice(
 				case client.Updates <- invoiceEvent.invoice:
 
 				case <-client.cancelChan:
+					i.deleteClient(client.id)
```

Comment:
Is there a reason to not just `defer i.deleteClient` when this goroutine exits? It'll mean that we also delete all our clients when we exit on `i.quit`, but I don't think that'll be a big hit, and it simplifies things a bit?

Reply:
Switched. Code-wise does simplify things, not necessarily logic-wise as it may increase the difficulty to follow. Will do a bit of refactoring when the time comes.